File: testsuite/results/systest2.nim
type
  TGtkMyStruct* {.importc: "GTK_MyStruct", header: "iup.h", bycopy.} = object
    a* {.importc: "a".}: mytype
    b* {.importc: "b".}: mytype

  PGtkMyStruct* = ptr TGtkMyStruct
  TGtkMyStruct* {.importc: "GTK_MyStruct", header: "iup.h", bycopy.} = object
    a* {.importc: "a".}: mytype
    b* {.importc: "b".}: mytype

  PGtkMyStruct* = ptr TGtkMyStruct

proc IupConvertXYToPos*(ih: PIhandle; x: cint; y: cint): cint {.cdecl,
    importc: "IupConvertXYToPos", header: "iup.h".}
proc handwrittenNim(): string =
  "@#"

const
  foobar* = 5 or 9

type
  wxEdge* {.size: sizeof(cint), pure.} = enum
    wxLeft, wxTop, wxRight, wxBottom, wxWidth, wxHeight, wxCentre, wxCentreX, wxCentreY

const
  wxCenter* = wxCentre

##  bug #136

proc bcf_float_set*(`ptr`: ptr cfloat; value: uint32_t) {.inline, cdecl.} =
  type
    INNER_C_UNION_systest2_48 {.importc: "no_name", header: "iup.h", bycopy, union.} = object
      i: uint32_t
      f: cfloat

  var u: INNER_C_UNION_systest2_48
  u.i = value
  `ptr`[] = u.f

proc sort*(a: ptr UncheckedArray[cint]; len: cint) {.cdecl, importc: "sort",
    header: "iup.h".}
##  bug #32

proc x*(): cint {.cdecl.} =
  when fii:
    if 1:
      discard
  else:
    return 1

File: testsuite/tests/systest2.c
#ifdef C2NIM
#  header "iup.h"
#  cdecl
#  mangle "'GTK_'{.*}" "TGtk$1"
#  mangle "'PGTK_'{.*}" "PGtk$1"
#endif

typedef struct stupidTAG {
  mytype a, b;
} GTK_MyStruct, *PGTK_MyStruct;

typedef struct  {
  mytype a, b;
} GTK_MyStruct, *PGTK_MyStruct;

int IupConvertXYToPos(PIhandle ih, int x, int y);

#def FOO_0()
#def FOO_1
#def FOO_2(x) x

FOO_0()
FOO_2(FOO_1)
FOO_1

#ifdef C2NIM
#@
proc handwrittenNim(): string =
  "@#"

@#
#endif

#define foobar #@ 5 or 9
@#

#pure wxEdge
enum wxEdge
{
    wxLeft, wxTop, wxRight, wxBottom, wxWidth, wxHeight,
    wxCentre, wxCenter = wxCentre, wxCentreX, wxCentreY
};

// bug #136

static inline void bcf_float_set(float *ptr, uint32_t value)
{
    union { uint32_t i; float f; } u;
    u.i = value;
    *ptr = u.f;
}

#isarray a

void sort(int* a, int len);

// bug #32

int x(){
#if fii
    if(1) {
    }
#else
    return 1;
#endif
}


#define someU64 12333ull

#define someU32 12333ul
#define someI64 12333LL
#define someI32 12333l
